Refactor route registration logic for improved

mdzahid-pro opened the pull request:
This commit enhances the route registration logic in the Laravel framework to support domain-specific API routes. It introduces a more streamlined approach to handle cases where a domain is provided, ensuring proper routing for API endpoints. The code has been refactored for improved readability and maintainability, offering a more robust solution for defining routes within different contexts.

<!--
Please only send a pull request to branches that are currently supported: https://laravel.com/docs/releases#support-policy 

If you are unsure which branch your pull request should be sent to, please read: https://laravel.com/docs/contributions#which-branch

Pull requests without a descriptive title, thorough description, or tests will be closed.

In addition, please describe the benefit to end users; the reasons it does not break any existing features; how it makes building web applications easier, etc.
-->

mdzahid-pro committed:
Refactor route registration logic for improved

This commit enhances the route registration logic in the Laravel framework to support domain-specific API routes. It introduces a more streamlined approach to handle cases where a domain is provided, ensuring proper routing for API endpoints. The code has been refactored for improved readability and maintainability, offering a more robust solution for defining routes within different contexts.

PerryvanderMeer commented on src/Illuminate/Foundation/Configuration/ApplicationBuilder.php:
```diff
@@ -274,7 +283,7 @@ public function withCommands(array $commands = [])
     {
         $this->app->afterResolving(ConsoleKernel::class, function ($kernel) use ($paths) {
-            $this->app->booted(fn () => $kernel->addCommandRoutePaths($paths));
+            $kernel->setCommandRoutePaths($paths);
```
Please undo these changes from #50742.

mdzahid-pro replied:
This is my first contribution in laravel if you can share more information how can which method should I undo that will help me to do this

mdzahid-pro committed:
This commit refactors the ApplicationBuilder class within the Illuminate\Foundation\Configuration namespace to improve its readability and maintainability. Key changes include:

Consistent method parameter naming and documentation.
Improved method signatures for better clarity and understanding.
Enhanced conditional logic within the withRouting method to streamline route configuration.
Improved type declarations and return types for method signatures.

mdzahid-pro committed:
removed return statement

mdzahid-pro committed:
styleci issue resolved

mdzahid-pro committed:
styleci test failed issue resolved

mdzahid-pro committed:
styleci test failed issue resolved

taylorotwell commented:
I am closing this pull request because it lacks sufficient explanation, tests, or both. It is difficult for us to merge pull requests without these things because the change may introduce breaking changes to the framework.

Feel free to re-submit your change with a **thorough** explanation of the feature and tests - integration tests are preferred over unit tests. Please include it's benefit to end users; the reasons it does not break any existing features; how it makes building web applications easier, etc.

Thanks!
